web/src/main/java/org/wso2/testgrid/web/api/SSOService.java

```diff
@@ -0,0 +1,153 @@
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509Credential;
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509KeyStoreCredential;
+import org.wso2.carbon.identity.sso.agent.saml.X509CredentialImpl;
+import org.wso2.carbon.identity.sso.agent.util.SSOAgentUtils;
+import org.wso2.testgrid.common.exception.TestGridException;
+import org.wso2.testgrid.common.util.TestGridUtil;
+import org.wso2.testgrid.web.bean.ErrorResponse;
+import org.wso2.testgrid.web.utils.ConfigurationContext;
+import org.wso2.testgrid.web.utils.Constants;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * REST Service implementation for SSO related functions.
+ */
+@Path("/acs")
+@Produces(MediaType.APPLICATION_JSON)
+public class SSOService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SSOService.class);
+
+    @javax.ws.rs.core.Context
+    ServletContext context;
+
+    @Context
+    HttpServletRequest request;
+
+    /**
+      * This has the implementation of the REST API for creating session and send the redirection information .
+      * @return Redirection to dashboard home page.
+     */
+    @POST
+    public Response createSession(@FormParam("SAMLResponse") String responseMessage) throws TestGridException {
+        XMLObject response;
+        org.opensaml.saml2.core.Response saml2Response;
+        try {
+            response = SSOAgentUtils.unmarshall(new String(Base64.decode(responseMessage), Charset.forName("UTF-8")));
+            saml2Response = (org.opensaml.saml2.core.Response) response;
+        } catch (SSOAgentException e) {
+            String msg = "Error occurred while unmarshalling SAMLResponse.";
+            logger.error(msg, e);
+            return Response.serverError().entity(
+                    new ErrorResponse.ErrorResponseBuilder().setMessage(msg + e.getMessage()).build()).build();
+        }
+
+        try {
+            validateSignature(saml2Response.getSignature());
+            request.getSession();
+            return Response.status(Response.Status.FOUND).
+                    header(HttpHeaders.LOCATION, Constants.WEBAPP_CONTEXT).
+                    type(MediaType.TEXT_PLAIN).build();
+        } catch (SSOAgentException e) {
+            String msg = "Observed an attempt with invalid SAMLResponse.";
+            logger.error(msg, e);
+            return Response.status(Response.Status.UNAUTHORIZED).entity(msg + e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Validates the XML Signature object
```

Comment:
Missing dot at the end. This doc does not properly explain what really happens in this method.